core/lib/types/src/fee_model.rs

```diff
@@ -0,0 +1,179 @@
+use core::panic;
+
+use serde::{Deserialize, Serialize};
+use zksync_system_constants::L1_GAS_PER_PUBDATA_BYTE;
+
+use crate::ProtocolVersionId;
+
+/// Fee input to be provided into the VM. It contains two options:
+/// - L1Pegged: L1 gas price is provided to the VM, and the pubdata price is derived from it. Using this option is required for the
+/// older versions of Era.
+/// - PubdataIndependent: L1 gas price and pubdata price are not necessarily dependend on one another. This options is more suitable for the
+/// newer versions of Era.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum BatchFeeInput {
```

Comment:
I'd suggest that we start with the PR that only introduces the BatchFeeInput (with L1Pegged). (as a part of breaking this large PR into pieces)

Reply:
Yes: https://github.com/matter-labs/zksync-era/pull/817/files